fili-core/src/main/java/com/yahoo/bard/webservice/config/CacheFeatureFlag.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Binds a caching strategy to a system configuration name.
+ */
+public enum CacheFeatureFlag implements FeatureFlag {
+    /**
+     * No cache.
+     */
+    NONE("query_response_caching_strategy", "NoCache"),
+    /**
+     * Use only TTL cache.
+     */
+    TTL("query_response_caching_strategy", "Ttl"),
+    /**
+     * Use only local signature cache.
+     */
+    LOCAL_SIGNATURE("query_response_caching_strategy", "LocalSignature"),
+    /**
+     * Use only etag cache.
+     */
+    ETAG("query_response_caching_strategy", "ETag");
+
+    private static final Logger LOG = LoggerFactory.getLogger(CacheFeatureFlag.class);
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+    private static final String QUERY_RESPONSE_CACHING_STRAGEGY = SYSTEM_CONFIG.getStringProperty(
+            SYSTEM_CONFIG.getPackageVariableName("query_response_caching_strategy"),
+            "None"
+    );
+
+    private final String propertyName;
```

Comment:
I'm going to change my vote on this, let's pull it out.  Clearly it's not needed now.